### PR TITLE
[#57] - Implement endpoints for liking/unliking a post

### LINKED ISSUE #57: Implement endpoints for liking/unliking a post
**Is your feature request related to a problem? Please describe.**
Currently we do not have endpoints for liking or unliking posts

**Describe the solution you'd like**
- Implement two endpoints: POST /posts/:id/like for liking a post (does not require a body) and DELETE /posts/:id/like for unliking a post. Those endpoints should automatically add/remove entries from the liked-posts table

Add tests for the two endpoints

### PR DESCRIPTION
**Pull request by LemonPepperSeasoning**
# Description:
Please provide a summary of changes made and the relevant issue that has been worked on. Also list any major changes made to project structure/dependencies if applicable

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [x] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: backend/__tests__/posts.js [hunk 1]
```diff
@@ -397,6 +452,64 @@ describe('Posts', () => {
             })
         })
 
+        describe('when user unlike post in a valid way', () => {
```
Comment (Dj5822) [inline backend/__tests__/posts.js]: Move all the tests that you added into a separate describe() where you indicate the path of the request.

**Review by Dj5822 — CHANGES_REQUESTED** [review]
Looks good, except I think you should organize the tests into a separate describe() function. Currently it looks like the new tests are all inside describe('GET /posts'). Make sure it's inside describe('POST /posts/:id/like').

File: backend/__tests__/posts.js (hunk 1 above)
Comment (LemonPepperSeasoning) [reply backend/__tests__/posts.js]: Thanks for the review and the comment. 
I have just made the change !
Let me know if you think it is ready to be merged

**Review by harristomy — APPROVED** [review]

File: backend/__tests__/posts.js [hunk 2]
```diff
@@ -402,4 +402,121 @@ describe('Posts', () => {
...
+                const user1 = await models.users.create({
+                    username: 'gandalf',
+                    nickname: 'gandalf',
+                    email: 'gandalf@gandalf.com',
+                    password: 'password',
+                })
+
+                const authToken = Authentication.generateAuthToken(user1)
+
+                const createPostResponse = await request(server)
```
Comment (Dj5822) [inline backend/__tests__/posts.js]: I made this same exact mistake, but we're not supposed to use other endpoints in the tests. What we have to do is to directly create a post by manipulating the model. You can check the updated version of my tests in my PR to get an idea of what I mean.

**Review by Dj5822 — CHANGES_REQUESTED** [review]
I realized that you made the same exact mistake that I did. It's not a major problem but it's probably best to fix it.

File: backend/__tests__/posts.js (hunk 2 above)
Comment (LemonPepperSeasoning) [reply backend/__tests__/posts.js]: Ok. 
Thanks for the great feedback !

File: backend/__tests__/posts.js [hunk 3]
```diff
@@ -402,4 +402,146 @@ describe('Posts', () => {
...
+                const newPost = await models.posts.create({
+                    text_content: 'Test text',
+                    author: user1.id,
+                    parent: null,
+                })
+
+                const invalidId = newPost.id + 999
+
+                const response = await request(server)
+                    .put(`/api/posts/${invalidId}/like`)
```
Comment (Dj5822) [inline backend/__tests__/posts.js]: I think this part is supposed to be post instead of put.

**Review by Dj5822 — CHANGES_REQUESTED** [review]
One last thing.

File: backend/__tests__/posts.js (hunk 3 above)
Comment (LemonPepperSeasoning) [reply backend/__tests__/posts.js]: Yes. you are completely right!.
Sorry about that.

**Review by Dj5822 — COMMENTED** [review]
LGTM

**Review by Dj5822 — APPROVED** [review]

**Pull request merged**